keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,1044 @@
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- `pkg/controller/job`: `2023-12-20` - `91.5%`
+- `pkg/registry/batch/job`: `2023-12-20` - `92.2%`
+- `pkg/apis/batch/v1`: `2023-12-20` - `29.3%` (mostly generated code)
+
+The following scenarios are covered:
+- the Job controller reconciles jobs with the `managed-by` label equal to `job-controller.k8s.io` when the feature is enabled
+- the Job controller reconciles jobs without the `managed-by` label when the feature is enabled
+- the Job controller does not reconcile jobs with custom value of the `managed-by` label when the feature is enabled
+- the Job controller reconciles jobs with custom `managed-by` label when the feature gate is disabled
+- verify the label is immutable, both when the job is suspended or unsuspended; when the feature is enabled
+- enablement / disablement of the feature after the Job (with custom `managed-by` label) is created
+
+##### Integration tests
+
+The following scenarios are covered:
+- the Job controller reconciles jobs with the `managed-by` label equal to `job-controller.k8s.io`
+- the Job controller reconciles jobs without the `managed-by` label
+- the Job controller does not reconcile a job with any other value of the `managed-by` label
+- the Job controller reconciles jobs with custom `managed-by` label when the feature gate is disabled
+
+During the implementation more scenarios might be covered.
+
+##### e2e tests
+
+The feature does not depend on kubelet, so the functionality can be fully
+covered with unit & integration tests.
+
+We propose a single e2e test for the following scenario:
+- the Job controller does not reconcile a job with any other value of the `managed-by` label
+
+### Graduation Criteria
+
+#### Beta
+
+- skip synchronization of jobs when the `managed-by` label does not exist, or equals `job-controller.k8s.io`
+- unit & integration tests
+- The feature flag enabled by default
+
+#### GA
+
+- Address reviews and bug reports from Beta users
+- e2e test
+- Re-evaluate the ideas of improving debuggability (like [extended `kubectl`](#debuggability), [dedicated condition](#condition-to-indicated-job-is-skipped), or [events](#event-indicating-the-job-is-skipped))
+- Re-evaluate the support for mutability of the label
+- Lock the feature gate
+
+#### Deprecation
+
+- Remove the feature-gate in GA+2.
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade
+
+An upgrade to a version which supports this feature does not require any
+additional configuration changes. This feature is opt-in at the Job-level, so
+to use it users need to add the `managed-by` label to their Jobs.
+
+#### Downgrade
+
+A downgrade to a version which does not support this feature does not require
```

Comment:
What about the downgrade to a version w/o this feature, ie. 1.29?

Reply:
This is what the paragraph was about, added a note in brackets about 1.29 or below specifically.

Also, mentioned the risk of running 2 controllers.